### PR TITLE
Prevent swingtimer from triggering when not in attacking state

### LINKED ISSUE #3178: Swing Timer starts on ending cast even when not currently attacking
### Description

When a cast ends the swing timer trigger is activated even when the character is not currently attacking.

### WeakAuras Version

WeakAuras 3.5.0

### World of Warcraft Flavor

The Burning Crusade

### Tested with only WeakAuras

- [X] Yes
- [ ] No

### Lua Error

_No response_

### Reproduction Steps

1. Cast something
2. See the swing timer activate

### Last Good Version

_No response_

### Screenshots

_No response_

### Export String

_No response_

### PR DESCRIPTION
**Pull request by Aviana**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes #3178

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- Tested in TBC client, avoided classic only function "IsPlayerAttacking" and used events to track attack status instead

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings

### CODE REVIEW

File: WeakAuras/GenericTrigger.lua [hunk 1]
```diff
@@ -1624,6 +1626,10 @@ do
         lastSwingOff, swingDurationOff = nil, nil
         WeakAuras.ScanEvents("SWING_TIMER_END")
       end
+    elseif event == "PLAYER_ENTER_COMBAT" then
+      isAttacking = true
```
Comment (InfusOnWoW) [inline WeakAuras/GenericTrigger.lua:1630]: That's obviously not correct. A player can enter combat without auto attacking.

Comment (Aviana) [reply WeakAuras/GenericTrigger.lua:1630]: See https://wowpedia.fandom.com/wiki/PLAYER_ENTER_COMBAT

Comment (InfusOnWoW) [reply WeakAuras/GenericTrigger.lua:1630]: Ah indeed.

**Review by mrbuds — APPROVED** [review]

**Comment by InfusOnWoW** [comment]
The commit message needs work though.

**Comment by Aviana** [comment]
> 
> 
> The commit message needs work though.

I don't know if that can be edited.

**Comment by mrbuds** [comment]
```
git commit --amend -m "Prevent swingtimer from triggering when not in attacking state, fixes #3179"
git push -f
```

**Pull request merged**